### ISSUE TITLE
Failures when running unittests on Arch Linux

### ISSUE THREAD

**Issue by jelly** [rogerbinns/apsw#205]
When running the testsuite on Arch Linux I get multiple tracebacks.

APSW version 3.9.2-r1
SQLite lib version 3.10.2
SQLite headers version 3010002

```
........................................................................Traceback (most recent call last):
  File "setup.py", line 884, in <module>
    'win64hackvars': win64hackvars}
  File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "setup.py", line 113, in run
    result=unittest.TextTestRunner(verbosity=self.show_tests+1).run(suite)
  File "/usr/lib/python2.7/unittest/runner.py", line 151, in run
    test(result)
  File "/usr/lib/python2.7/unittest/suite.py", line 70, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python2.7/unittest/suite.py", line 108, in run
    test(result)
  File "/usr/lib/python2.7/unittest/suite.py", line 70, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python2.7/unittest/suite.py", line 108, in run
    test(result)
  File "/usr/lib/python2.7/unittest/case.py", line 393, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/jelle/projects/community/python-apsw/trunk/src/apsw2/tests.py", line 4757, in testVFS
    testdb()
  File "/home/jelle/projects/community/python-apsw/trunk/src/apsw2/tests.py", line 8093, in testdb
    db.cursor().execute("create table foo(x,y); insert into foo values(1,2); insert into foo values(date('now'), date('now'))")
  File "src/vfs.c", line 1114, in vfs.xCurrentTime
    AddTraceBackHere(__FILE__, __LINE__, "vfs.xCurrentTime", "{s: O}", "result", pyresult?pyresult:Py_None);
  File "src/pyutil.c", line 158, in Call_PythonMethod
    AddTraceBackHere(__FILE__, __LINE__, "Call_PythonMethod", "{s: s, s: i, s: O, s: O}",
apsw.VFSNotImplementedError: VFSNotImplementedError: Method xCurrentTime is not implemented
Traceback (most recent call last):
  File "setup.py", line 884, in <module>
    'win64hackvars': win64hackvars}
  File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "setup.py", line 113, in run
    result=unittest.TextTestRunner(verbosity=self.show_tests+1).run(suite)
  File "/usr/lib/python2.7/unittest/runner.py", line 151, in run
    test(result)
  File "/usr/lib/python2.7/unittest/suite.py", line 70, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python2.7/unittest/suite.py", line 108, in run
    test(result)
  File "/usr/lib/python2.7/unittest/suite.py", line 70, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python2.7/unittest/suite.py", line 108, in run
    test(result)
  File "/usr/lib/python2.7/unittest/case.py", line 393, in __call__
    return self.run(*args, **kwds)
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/jelle/projects/community/python-apsw/trunk/src/apsw2/tests.py", line 4757, in testVFS
    testdb()
  File "/home/jelle/projects/community/python-apsw/trunk/src/apsw2/tests.py", line 8093, in testdb
    db.cursor().execute("create table foo(x,y); insert into foo values(1,2); insert into foo values(date('now'), date('now'))")
  File "src/vfs.c", line 1114, in vfs.xCurrentTime
    AddTraceBackHere(__FILE__, __LINE__, "vfs.xCurrentTime", "{s: O}", "result", pyresult?pyresult:Py_None);
  File "src/pyutil.c", line 158, in Call_PythonMethod
    AddTraceBackHere(__FILE__, __LINE__, "Call_PythonMethod", "{s: s, s: i, s: O, s: O}",
apsw.VFSNotImplementedError: VFSNotImplementedError: Method xCurrentTime is not implemented
```

**Comment by rogerbinns** [comment]
This was due to an incompatible change in SQLite.  The SQLite authors have fixed it in 3.11.0.  SQLite 3.11.0 will be released in less than two weeks which is why there is no APSW 3.10 release.

The current version of tests.py also works around the issue.

BTW why is the directory name  apsw**2**?

**Comment by jelly** [comment]
Thanks for the information! the directory was called apsw2 since it's for the python2 build :)
